scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
 -----------------------------------
 ---@type TMobEntity
 local entity = {}
+local ID = zones[xi.zone.VELUGANNON_PALACE]
 
-local pathNodes =
+local firstBlueDoorCoord = { x = -160, y = -2.974, z = 380 }
+local middleBlueDoorCoord = { x = -80, y = -2.974, z = 460 }
+local middleYellowDoorCoord = { x = 80, y = -2.974, z = 460 }
+local firstYellowDoorCoord = { x = 160, y = -2.974, z = 380 }
```

Comment:
Do you need these, if you're looking up the door entities themselves? You can just use `getPos()` on them, and `checkDistance()` for comparisons

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
Swapped these to `GetNPCByID` and then using `getPos()` to determine their locations instead of hard-coding them